src/test/java/picard/analysis/CollectWgsMetricsTestUtils.java

```diff
@@ -96,6 +118,97 @@ private static void createTestSAM(String testSamName) throws IOException {
         writer.close();
     }
 
+    static IntervalList createIntervalList() {
+        return new IntervalList(new SAMFileHeader());
+    }
+
+    static AbstractLocusIterator createReadEndsIterator(String exampleSam){
+        final List<SamRecordFilter> filters = new ArrayList<>();
+        final CountingFilter dupeFilter = new CountingDuplicateFilter();
+        final CountingFilter mapqFilter = new CountingMapQFilter(0);
+        filters.add(new SecondaryAlignmentFilter()); // Not a counting filter because we never want to count reads twice
+        filters.add(mapqFilter);
+        filters.add(dupeFilter);
+        SamReader samReader = createSamReader(exampleSam);
+        AbstractLocusIterator iterator =  new EdgeReadIterator(samReader);
+        iterator.setSamFilters(filters);
+        iterator.setMappingQualityScoreCutoff(0); // Handled separately because we want to count bases
+        iterator.setIncludeNonPfReads(false);
+        return iterator;
+    }
+
+    static SamReader createSamReader(String samExample) {
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(samExample.getBytes());
+        return SamReaderFactory.makeDefault().open(SamInputResource.of(inputStream));
+    }
+
+    static ReferenceSequenceFileWalker getReferenceSequenceFileWalker(String referenceString){
+        ReferenceSequenceFile referenceSequenceFile = createReferenceSequenceFile(referenceString);
+        return new ReferenceSequenceFileWalker(referenceSequenceFile);
+    }
+
+    static ReferenceSequenceFileWalker getReferenceSequenceFileWalker(){
+        String referenceString = ">ref\nACCTACGTTCAATATTCTTC";
+        return getReferenceSequenceFileWalker(referenceString);
+    }
+
+    static ReferenceSequenceFile createReferenceSequenceFile(String referenceString) {
```

Comment:
InMemoryReferenceSequence in htsjdk semmes to have a package-private access level and as we can see is used only for testing. Should we rewrite it in picard tools?